### PR TITLE
[categorical] Simplify and clean up

### PR DESCRIPTION
**Pull request by tkoeppe**

### CODE REVIEW

File: distributions/categorical.lua [hunk 1]
```diff
@@ -10,46 +10,46 @@ function distributions.cat.logpdf(...)
...
         options = {}
+        I = _idxResult(1)
     elseif nArgs == 2 then
         -- (N,  p) or (p, options)
         if type(select(2, ...)) == 'table' then
             -- (p, options)
             p, options = ...
             options = options or {}
-            I, N = _idxResultAndN(1, options.categories)
+            I = _idxResult(1)
```
Comment (d11) [inline distributions/categorical.lua]: is removing options.categories here deliberate?

File: distributions/categorical.lua [hunk 2]
```diff
@@ -10,46 +10,46 @@ function distributions.cat.logpdf(...)
     error('Not implemented')
 end
 
-local function _idxResultAndN(x, categories)
+local function _idxResult(x, categories)
     -- return res, N
```
Comment (d11) [inline distributions/categorical.lua]: comment is no longer accurate

File: distributions/categorical.lua (hunk 1 above)
Comment (tkoeppe) [inline distributions/categorical.lua]: Yes, as `_idxResult` does not consume the second argument if the first one has type `number`.

File: distributions/categorical.lua (hunk 2 above)
Comment (tkoeppe) [inline distributions/categorical.lua]: Yes, indeed. Deleted.

**Pull request merged**